src/Psalm/Internal/PhpVisitor/Reflector/ClassLikeNodeScanner.php

```diff
@@ -1345,10 +1346,41 @@ private function visitEnumDeclaration(
         $case_location = new CodeLocation($this->file_scanner, $stmt);
 
         if (!isset($storage->enum_cases[$stmt->name->name])) {
-            $storage->enum_cases[$stmt->name->name] = new EnumCaseStorage(
+            $case = new EnumCaseStorage(
                 $enum_value,
                 $case_location
             );
+
+            $attrs = $this->getAttributeStorageFromStatement(
+                $this->codebase,
+                $this->file_scanner,
+                $this->file_storage,
+                $this->aliases,
+                $stmt,
+                $this->storage->name ?? null
+            );
+
+            foreach ($attrs as $attribute) {
+                if ($attribute->fq_class_name === 'Psalm\\Deprecated'
+                    || $attribute->fq_class_name === 'JetBrains\\PhpStorm\\Deprecated'
+                ) {
+                    $case->deprecated = true;
+                }
+            }
+
+            $comment = $stmt->getDocComment();
+            if ($comment) {
+                $var_comments = CommentAnalyzer::getTypeFromComment(
```

Comment:
This doesn't sound right. Did you mean to use `DocblockParser::parsePreservingLength()`, like it's used for constants here?: https://github.com/vimeo/psalm/blob/646ba983a0fb09a4ed4946ac2fc47ac192a111d1/src/Psalm/Internal/PhpVisitor/Reflector/ClassLikeNodeScanner.php#L1207-L1211

Reply:
ah! Much better. I wasn't aware of that one.

Fixed.